src/lib/config.coffee

```diff
@@ -123,46 +111,28 @@ module.exports = (dnschain) ->
 
     nconf.file 'global', {file:"/etc/#{appname}/#{appname}.conf", format:props}
 
-    # TODO: this blockchain-specific config stuff should be moved out of
-    #       this file and into the constructors (NMCPeer, BDNSPeer, etc.)
-
-    # namecoin
-    nmc = (new nconf.Provider()).argv().env()
-    
-    # TODO: use the same _.find technique as done below for bdnsConf
-    nmcConf = if process.env.APPDATA?
-        path.join process.env.APPDATA, "Namecoin", "namecoin.conf"
-    else if process.env.HOME?
-        path.join process.env.HOME, ".namecoin", "namecoin.conf"
-
-    nmc.file('user', {file:nmcConf,format:props}) if nmcConf
-
-    # bdns
-    bdns = (new nconf.Provider()).argv().env()
-    bdnsConf = _.find _.filter([
-        [process.env.APPDATA, "KeyID"],
-        [process.env.HOME, ".KeyID"],
-        [process.env.HOME, "Library", "Application Support", "KeyID"]]
-    , (x) -> !!x[0])
-    , (x) -> fs.existsSync path.join x...
-
-    if bdnsConf
-        bdns.file 'user', file: path.join(bdnsConf..., 'config.json')
-    else
-        # TODO: this!
-
     stores =
         dnschain: nconf.defaults defaults
-        nmc: nmc.defaults nmcDefs
-        bdns: bdns.defaults bdnsDefs
 
     config =
         get: (key, store="dnschain") -> stores[store].get key
         set: (key, value, store="dnschain") -> stores[store].set key, value
-        # Namecoin's config is not ours, so we don't pretend it is
-        nmc:
-            get: (key)-> config.get key, 'nmc'
-            set: (key, value)-> config.set key, value, 'nmc'
-        bdns:
-            get: (key)-> config.get key, 'bdns'
-            set: (key, value)-> config.set key, value, 'bdns'
+        add: (name, path) ->
+            if stores[name]?
+                return
+            path = [path] if not Array.isArray(path)
+            path.push(stores.dnschain.get("#{name}:config")) if stores.dnschain.get("#{name}:config")?
+            conf = (new nconf.Provider()).argv().env()
+            confFile = _.find path, (x) -> fs.existsSync x
+            conf.file(confFile) if confFile
+            conf.defaults(stores.dnschain.get("#{name}")) if stores.dnschain.get("#{name}")?
+            stores[name] = conf
+            config[name] = conf
+        hosts: ->
+            _.uniq [
+                "127.0.0.", "10.0.0.", "192.168.", "::1", "fe80::"
+                gConf.get('dns:host'), gConf.get('http:host')
+                gConf.get('dns:externalIP'), gExternalIP()
+                _.map(stores, (c) ->
+                    c.get('host'))...
+            ].filter (o)-> typeof(o) is 'string'
```

Comment:
OK, you can keep it here since it's semi-configurable (since it's referencing the config vars), however:

1. Needs to be renamed as `localhosts:`
2. Pretty sure you can't use `gConf`, it will be undefined. You can use `config.get` instead of `gConf.get`